machines/Makefile

```diff
@@ -1,50 +1,44 @@
-find-dir-with = $(shell /usr/bin/perl -e 'chomp($$_ = `pwd`); while ($$_ ne "" && ! -e "$$_/$(1)") { m:(.*)/[^/]+/??:; $$_ = $$1; } print;')
-
-ifndef BSG_MANYCORE_DIR
-  export BSG_MANYCORE_DIR := $(call find-dir-with,.BSG_MANYCORE_ROOT)
-endif
+.DEFAULT_GOAL = all
 
+export BSG_MANYCORE_DIR := $(shell git rev-parse --show-toplevel)
 export BASEJUMP_STL_DIR := $(BSG_MANYCORE_DIR)/../basejump_stl
 
 # Sets "VCS" variable
 include $(BSG_MANYCORE_DIR)/../bsg_cadenv/cadenv.mk
 
-MACHINES = 9x15 4x4
-
-SIMV = $(addsuffix /simv, $(MACHINES))
-SIMV_DEBUG = $(addsuffix /simv-debug, $(MACHINES))
-SIMV_DAIDIR = $(addsuffix .daidir, $(SIMV) $(SIMV_DEBUG))
+# Include source lists
+include Makefile.include
 
-VCS_INCDIR := +incdir+$(BASEJUMP_STL_DIR)/bsg_misc
-VCS_INCDIR += +incdir+$(BASEJUMP_STL_DIR)/bsg_cache
-VCS_INCDIR += +incdir+$(BASEJUMP_STL_DIR)/bsg_noc
-VCS_INCDIR += +incdir+$(BSG_MANYCORE_DIR)/v
-VCS_INCDIR += +incdir+$(BSG_MANYCORE_DIR)/v/vanilla_bean
+MACHINES = 9x15 4x4
 
 .PHONY: all clean
 
-all: $(SIMV)
+all: $(foreach machine, $(MACHINES), $(machine)/stamp)
 
-%/simv:
+%/stamp: $(VCS_SOURCES) $(VCS_INCLUDES)
 	$(eval include $*/Makefile.machine.include)
-	$(eval VCS_DEFINE := +define+BSG_MACHINE_GLOBAL_X=${BSG_MACHINE_GLOBAL_X})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_GLOBAL_Y=${BSG_MACHINE_GLOBAL_Y})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_VCACHE_SET=${BSG_MACHINE_VCACHE_SET})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_VCACHE_WAY=${BSG_MACHINE_VCACHE_WAY})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_VCACHE_BLOCK_SIZE_WORDS=${BSG_MACHINE_VCACHE_BLOCK_SIZE_WORDS})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_DRAM_SIZE_WORDS=${BSG_MACHINE_DRAM_SIZE_WORDS})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_DRAM_INCLUDED=${BSG_MACHINE_DRAM_INCLUDED})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_MAX_EPA_WIDTH=${BSG_MACHINE_MAX_EPA_WIDTH})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_MEM_CFG=${BSG_MACHINE_MEM_CFG})
+	$(eval VCS_MACRO_DEFS := +define+BSG_MACHINE_GLOBAL_X=${BSG_MACHINE_GLOBAL_X})
```

Comment:
Closer! What I meant was something like: 
```
	VDEFINES += BSG_MACHINE_GLOBAL_Y=${BSG_MACHINE_GLOBAL_Y})
	VDEFINES += BSG_MACHINE_VCACHE_SET=${BSG_MACHINE_VCACHE_SET})
        # etc...
  
       VCS_DEFINES +: $(foreach def,$(VDEFINES),+define+"$(def)")

```

Reply:
We have two types of `.v` files in basejump_stl. `bsg_defines.v` is one kind with defines and `bsg_dff_reset.v` is the other kind with actual rtl. Hence, I used `VDEFINES` for `.v` with defines and `MACRO_DEFS` for cmdline macro defs. 

I suggest you include this distinction between sources as well in F1 because, this way we can group `VDEFINES` and make sure to compile them first.